native/src/main/java/io/ballerina/lib/data/jsondata/json/JsonCreator.java

```diff
@@ -395,6 +395,9 @@ static Type getMemberType(Type expectedType, int index, boolean allowDataProject
         }
 
         Type referredType = TypeUtils.getReferredType(expectedType);
+        if (referredType.getTag() == TypeTags.INTERSECTION_TAG) {
+            return getMemberType(((IntersectionType) expectedType).getEffectiveType(), index, allowDataProjection);
```

Comment:
Is this correct? You're checking if `referredType` is an intersection type and casting `expectedType` as an `IntersectionType`?

Reply:
Fixed in 
https://github.com/ballerina-platform/module-ballerina-data.jsondata/pull/45/commits/5dcb0adf370561e947623781778fad86cb318421